generate.py

```diff
@@ -113,11 +115,16 @@
                 "parent": "GC1"
             }
 
-    for idx, capacity in enumerate(args.battery or []):
+    for idx, (capacity, c_rate) in enumerate(args.battery):
+        if capacity > 0:
+            max_power = c_rate * capacity
+        else:
+            # unlimited battery: set power directly
+            max_power = c_rate
         batteries["BAT{}".format(idx+1)] = {
             "parent": "GC1",
             "capacity": capacity,
-            "charging_curve": [[0, 50], [80, 50], [100, 0]]
+            "charging_curve": [[0, max_power], [100, max_power]]
```

Comment:
for now totally fine. Perspective could be to include real charging curves of stationary batteries